solutions/usaco-1060.mdx

```diff
@@ -53,5 +53,77 @@ int main(){
 ```
 
 </CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+	
+class Main {
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
```

Comment:
scanner should be avoided https://usaco.guide/general/adding-solution?lang=java#java can you change it to kattio?

Reply:
i mean it still get's ac'd w/o

Reply:
no scanners 👀

Reply:
also rename to `io` after changing to kattio

Reply:
ye this is bronze theres no way scanner effects run time in bronze lmfao 

but again maggie _is_ content author so I should listen to her. I will change that thing if another person tells me to 👍

Reply:
so is kattio this whole thing
```java
class Kattio extends PrintWriter {
	private BufferedReader r;
	private StringTokenizer st;
	// standard input
	public Kattio() { this(System.in,System.out); }
	public Kattio(InputStream i, OutputStream o) {
		super(o);
		r = new BufferedReader(new InputStreamReader(i));
	}
	// USACO-style file input
	public Kattio(String problemName) throws IOException {
		super(new FileWriter(problemName+".out"));
		r = new BufferedReader(new FileReader(problemName+".in"));
	}
	// returns null if no more input
	public String next() {
		try {
			while (st == null || !st.hasMoreTokens())
				st = new StringTokenizer(r.readLine());
			return st.nextToken();
		} catch (Exception e) {}
		return null;
	}
	public int nextInt() { return Integer.parseInt(next()); }
	public double nextDouble() { return Double.parseDouble(next()); }
	public long nextLong() { return Long.parseLong(next()); }
}
```
so I keep this thing in my code?

Reply:
Use the kattio [codesnip](https://usaco.guide/general/working-mdx#kattio)

Reply:
"kattio" more like "catgirl owo" isnt that right @SansPapyrus683

Reply:
@maggie-j-liu thanks for the snip ill look into it later tonight when I get the chance

Reply:
> "kattio" more like "catgirl owo" isnt that right @SansPapyrus683

you better sleep with one eye open buddy

Reply:
ez

Reply:
can we not lmao.

Reply:
done kattio... it doesnt come in github so all it says is "//CodeSnip{Kattio}	" but when you paste my code into usaco guide editor it works 
😄